cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -280,11 +278,19 @@ public void resume() throws Exception {
 
   @Override
   protected void shutDown() throws Exception {
-    httpService.stop();
-    destroyWorkflow();
-    deleteLocalDatasets();
-    if (pluginInstantiator != null) {
-      pluginInstantiator.close();
+    // Clear the interrupt flag.
```

Comment:
what's the impact of not doing this commit?

Reply:
The WorkflowDriver, which is a guava execution service, uses the same thread for the start, run and stop methods. Since in triggetShutdown it interrupts the run thread and the run thread may or may not be clearing that (depending on what was executing during the interruption). If the thread interrupt flag was set, calling stop on the netty http service would throw an interrupted exception (it is blocking to wait for threads to shutdown), which is not desired.